tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py

```diff
@@ -109,94 +141,206 @@ def test(self):
         self.simulator.go(5)
         self.assertEqual(self.nodes[MED1].get_state(), 'child')
 
-        # 3 MED1: MED1 sends an ICMPv6 Echo Request to Router1 using GUA 2003::
-        # address
-        router1_addr = self.nodes[ROUTER1].get_addr("2003::/64")
+        self.collect_rlocs()
+        self.collect_rloc16s()
+        self.collect_ipaddrs()
+
+        # 3 MED1: MED1 sends an ICMPv6 Echo Request to Router1 using GUA
+        # PREFIX_1 address
+        router1_addr = self.nodes[ROUTER1].get_addr(PREFIX_1)
         self.assertTrue(router1_addr is not None)
         self.assertTrue(self.nodes[MED1].ping(router1_addr))
 
         # Wait for sniffer got Address Notification messages
         self.simulator.go(1)
```

Comment:
It might not be necessary to wait here with the new test verification approach.

Reply:
keep waiting 1 sec for preventing packets disorder though it is quite slim possibility